youtube_dl/extractor/tiktok.py

```diff
@@ -1,24 +1,63 @@
 # coding: utf-8
 from __future__ import unicode_literals
 
+import re
+
 from .common import InfoExtractor
+from .generic import GenericIE
+from ..compat import (
+    compat_kwargs,
+)
 from ..utils import (
-    compat_str,
+    dict_get,
     ExtractorError,
     float_or_none,
+    get_element_by_id,
     int_or_none,
     str_or_none,
     try_get,
     url_or_none,
 )
 
 
+# decorator enforces UA that TT doesn't block
+def vanilla_UA_request(func):
+
+    vanilla_UA = 'Mozilla/5.0'
+
+    def wrapped(*args, **kwargs):
+        headers = kwargs.get('headers', {})
+        if 'User-Agent' not in headers:
+            headers['User-Agent'] = vanilla_UA
+            kwargs.update({'headers': headers, })
+            kwargs = compat_kwargs(kwargs)
+        return func(*args, **kwargs)
+
+    return wrapped
+
+
 class TikTokBaseIE(InfoExtractor):
+    IE_DESC = 'Abstract base for TikTok extractors'
+    IE_NAME = 'tiktok:base'
+
+    @vanilla_UA_request
+    def _request_webpage(self, *args, **kwargs):
+        return super(TikTokBaseIE, self)._request_webpage(*args, **kwargs)
+
+    def _get_SIGI_STATE(self, video_id, html):
+        state = self._parse_json(
+            get_element_by_id('SIGI_STATE', html)
+            or self._search_regex(
+                r'''(?s)<script\s[^>]*?\bid\s*=\s*(?P<q>"|'|\b)sigi-persisted-data(?P=q)[^>]*>[^=]*=\s*(?P<json>{.+?})\s*(?:;[^<]+)?</script''',
```

Comment:
```suggestion
                 r'''(?s)<script\s[^>]*?\bid\s*=\s*(?P<q>"|'|\b)(sigi-persisted-data|SIGI_STATE)(?P=q)[^>]*>[^=]*?=?\s*(?P<json>{.+?})\s*(?:;[^<]+)?</script''',
```
id has changed in html and no longer assigned to variable.

Reply:
can @dirkf review this?

Reply:
Check the new line `get_element_by_id('SIGI_STATE', html)` above which fixes that issue.